docs/sources/service.md

```diff
@@ -0,0 +1,108 @@
+# Ingress source
+
+The service source creates DNS entries based on `Service` resources.
+
+## Filtering the Services considered
+
+The `--service-type-filter` flag filters Service resources by their `spec.type`.
+The flag may be specified multiple times to allow multiple service types.
+
+This source supports the `--label-filter` flag, which filters Service resources
+by a set of labels.
+
+## Domain names
+
+The domain names of the DNS entries created from a Service are sourced from the following places:
+
+* Adds the domain names from any `external-dns.alpha.kubernetes.io/hostname` and/or
+`external-dns.alpha.kubernetes.io/internal-hostname` annotation.
+This behavior is suppressed if the `--ignore-hostname-annotation` flag was specified.
+
+* If no DNS entries were produced for a Service by the previous steps
+and the `--compatibility` flag was specified, then adds DNS entries per the
+selected compatibility mode.
+
+* If no DNS entries were produced for a Service by the previous steps
+or the `--combine-fqdn-annotation` flag was specified, then adds domain names
+generated from any`--fqdn-template` flag.
+
+### Domain names for headless service pods
+
+If a headless service (without an `external-dns.alpha.kubernetes.io/target` annotation) creates DNS entries with targets from
+a Pod that has a non-empty `spec.hostname` field, additional DNS entries are created for that Pod, containing the targets from that Pod.
+For each domain name created for the Service, the additional DNS entry for the Pod has that domain name prefixed with
+the value of the Pod's `spec.hostname` field and a `.`.
+
+## Targets
+
+If the Service has an `external-dns.alpha.kubernetes.io/target` annotation, uses 
+the values from that. Otherwise, the targets of the DNS entries created from a service are sourced depending
+on the Service's `spec.type`:
+
+### LoadBalancer
+
+* If the hostname came from an `external-dns.alpha.kubernetes.io/internal-hostname` annotation, uses
+the Service's `spec.clusterIP` field. If that field has the value `None`, does not generate
+any targets for the hostname.
+
+* Otherwise, if the Service has one or more `spec.externalIPs`, uses the values in that field.
+
+* Otherwise, iterates over each `status.loadBalancer.ingress`, adding any non-empty `ip` and/or `hostname`.
+If the `--resolve-service-load-balancer-hostname` flag was specified, any non-empty `hostname`
+is instead queried through DNS and any resulting IP addresses are added instead.
+A DNS query failure results in zero targets being added for that load balancer's ingress hostname.
```

Comment:
```suggestion
* Otherwise, iterates over each `status.loadBalancer.ingress`, adding any non-empty `ip` and/or `hostname`.

If the `--resolve-service-load-balancer-hostname` flag was specified, any non-empty `hostname`
is queried through DNS and any resulting IP addresses are added instead.
A DNS query failure results in zero targets being added for that load balancer's ingress hostname.
```